In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from scipy.stats import norm
plt.rcParams['figure.figsize'] = (10,5)
plt.rcParams['figure.dpi'] = 300
sns.set_theme(style='darkgrid', palette='rainbow')
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline


## Hypothesis Testing

### Question 1

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [2]:
df =pd.read_csv(r"C:\Users\shree\Desktop\subject\Data_Science\DS_Assignments\Datasets\Cutlets.csv")

In [3]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
df.shape

(35, 2)

In [5]:
Unit_A = df['Unit A']

In [6]:
Unit_B = df['Unit B']

s = 35
signaficance leval  = 5%
α = 0.05

Step 1 : Define null and alternate hypothesis

H0(null Hypothesis) : Unit A = Unit B
Ha(Alternate Hypothesis) : Unit A != Unit B

Step 2 : select the test from Z-test or t test

Here we dont have population parameters so selecting T-test

Step 3 :

selection of one tail or two tail test

Inputs are Diameter 1 for unit A and and Diameter 2 for unit B that is Discrete in two categories
Output is continuous as we are trying to see the difference in diameter and diameter is a continuous variable
so selcting 2 tailed t test

Step 4 :

Calculate α (significance leval)
α = 0.05

Step 5 : 

Calculate p value

In [7]:
p_value=stats.ttest_ind(Unit_A,Unit_B)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [8]:
p_value[1]     # 2-tail probability 

0.4722394724599501

Step 6 :

Accepting or rejecting Null Hypothesis

Comparing  p_value with α = 0.05 (At 5% significance level) , as p < α rejecting the null hypothesis 
hence we can conclude that there is significant difference between the diameter of Unit A and Unit B cutlets

### Question 2 

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. 
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [9]:
df1 = pd.read_csv(r"C:\Users\shree\Desktop\subject\Data_Science\DS_Assignments\Datasets\LabTAT.csv")

In [10]:
df1.shape

(120, 4)

In [11]:
df1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [12]:
Laboratory_1 = df1['Laboratory 1']

In [13]:
Laboratory_2 = df1['Laboratory 2']

In [14]:
Laboratory_3 = df1['Laboratory 3']

In [15]:
Laboratory_4 = df1['Laboratory 4']

s = 120
signaficance leval  = 5%
α = 0.05

Step 1 : Define null and alternate hypothesis

H0(null Hypothesis) : Laboratory_1 = Laboratory_2 = Laboratory_3 = Laboratory_4
Ha(Alternate Hypothesis) : Any of the laboratory TAT is not equal to other laboratory TAT

Step 2 : select the test 

Here we have more than 2 samples then we have to select Annova Test 

Step 3 :

Calculate f and p 
α = 0.05


In [16]:
f , p = stats.f_oneway(Laboratory_1,Laboratory_2,Laboratory_3,Laboratory_4)

In [17]:
if p < 0.05:
    print("reject null hypothesis , hence at least one sample is different")
else:
    print("Accept Null hypothesis i.e. all samples are same")

reject null hypothesis , hence at least one sample is different


### Question 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar 
across regions.

       East west North South  
Male   50   142   131    70

Female 550  351   480    350

H0       All proportions are equal

Ha       Not all Proportions are equal

Check p-value
If p-Value < alpha, we reject Null Hypothesis



In [18]:
df2 = pd.read_csv(r"C:\Users\shree\Desktop\subject\Data_Science\DS_Assignments\Datasets\BuyerRatio.csv")

In [19]:
df2.shape

(2, 5)

In [20]:
df2

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750


Confidance level  = 95%
hence,
α = 0.05

Step 1 : Define null and alternate hypothesis

H0(null Hypothesis) : All proportions are equal

Ha(Alternate Hypothesis) : Not all Proportions are equal

Step 2 : select the test

Here we have given  two Catogarical columns so selecting Chi-2 test

Step 3 :

Calculate α (significance leval)
Here confidance level is not given so considering 95% confidance level and for this leval 
α = 0.05

Step 4 : 

Calculate p value

In [22]:
alpha=0.05
Male = [50,142,131,70]
Female=[435,1523,1356,750]
Sales=[Male,Female]
print(Sales)

[[50, 142, 131, 70], [435, 1523, 1356, 750]]


In [23]:
chiStats = stats.chi2_contingency(Sales)
print('Test-t = %f, p-value = %f' % (chiStats[0], chiStats[1]))
print('Interpret by p-Value')
if chiStats[1] < 0.05:
  print('we reject null hypothesis')
else:
  print('we accept null hypothesis')

Test-t = 1.595946, p-value = 0.660309
Interpret by p-Value
we accept null hypothesis


In [24]:
#critical value = 0.1
alpha = 0.05
critical_value = stats.chi2.ppf(q = 1 - alpha,df=chiStats[2])# Find the critical value for 95% confidence degree of freedom

observed_chi_val = chiStats[0]
print('Interpret by critical value')
if observed_chi_val <= critical_value:
    # observed value is not in critical area therefore we accept null hypothesis
    print ('Accept the Null Hypothesis (All proportional are Equal)')
else:
    # observed value is in critical area therefore we reject null hypothesis
    print ('Failed to reject the Null Hypothesis (Not all Proportions are equal)')

Interpret by critical value
Accept the Null Hypothesis (All proportional are Equal)


### Question 4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [25]:
df3 = pd.read_csv(r"C:\Users\shree\Desktop\subject\Data_Science\DS_Assignments\Datasets\CustomerOrderForm.csv")

In [26]:
df3.shape

(300, 4)

In [27]:
df3.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

Significance level  = 95%
hence,
α = 0.05

Step 1 : Define null and alternate hypothesis

H0(null Hypothesis) : All proportions are equal

Ha(Alternate Hypothesis) : Not all Proportions are equal

Step 2 : select the test

Here we have given proportion of two Catogarical columns so selecting Chi-2 test

Step 3 :

Calculate α (significance leval)

α = 0.05

Step 4 : 

Calculate p value

In [28]:
Phillippines_value = df3['Phillippines'].value_counts()
Indonesia_value = df3['Indonesia'].value_counts()
Malta_value = df3['Malta'].value_counts()
India_value = df3['India'].value_counts()
print(Phillippines_value)
print(Indonesia_value)
print(Malta_value)
print(India_value)

Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Error Free    269
Defective      31
Name: Malta, dtype: int64
Error Free    280
Defective      20
Name: India, dtype: int64


In [29]:
chiStats = stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
print('Test t=%f, p-value=%f' % (chiStats[0], chiStats[1]))
print('Interpret by p-Value')
if chiStats[1] < 0.05:
  print('we reject null hypothesis')
else:
  print('we accept null hypothesis')

Test t=3.858961, p-value=0.277102
Interpret by p-Value
we accept null hypothesis


In [30]:
#critical value = 0.1
alpha = 0.05
critical_value = stats.chi2.ppf(q = 1 - alpha,df=chiStats[2])
observed_chi_val = chiStats[0]
print('Interpret by critical value')
if observed_chi_val <= critical_value:
       print ('Null hypothesis cannot be rejected (variables are not related)')
else:
       print ('Null hypothesis cannot be excepted (variables are not independent)')

Interpret by critical value
Null hypothesis cannot be rejected (variables are not related)
